File: Two_Sum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return False
        dict = {}
        for i in range(len(nums)):
            if nums[i] in dict:
                return [dict[nums[i]], i]
            else:
                dict[target - nums[i]] = i

if __name__ == "__main__":
    solution = Solution()
    nums = [ 2, 3, 4 ]
    for element in solution.twoSum(nums, 6):
        print element,

File: Two_Sum/twoSum.cpp
#include <iostream>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target -nums[i];
            if (hash.find(complement) != hash.end()) {
                return vector<int> {hash[complement], i};
            }
            hash[nums[i]] = i;
        }
        throw std::invalid_argument("no two sum solution");
    }
};

int main() {
    Solution solution;
    vector<int> vec = { 2, 3, 4};
    std::cout << "result: " << std::endl;
    for (auto v : solution.twoSum(vec, 6)) {
        std::cout << v << " ";
    }
}
